packages/clerk-js/src/core/resources/OrganizationDomain.ts

```diff
@@ -0,0 +1,88 @@
+import type {
+  OrganizationDomainJSON,
+  OrganizationDomainResource,
+  OrganizationDomainVerification,
+  OrganizationEnrollmentMode,
+  UpdateOrganizationDomainParams,
+} from '@clerk/types';
+
+import { unixEpochToDate } from '../../utils/date';
+import { BaseResource } from './Base';
+
+export class OrganizationDomain extends BaseResource implements OrganizationDomainResource {
+  id!: string;
+  name!: string;
+  organizationId!: string;
+  enrollmentMode!: OrganizationEnrollmentMode;
+  verification!: OrganizationDomainVerification | null;
+  affiliationEmailAddress!: string | null;
+  createdAt!: Date;
+  updatedAt!: Date;
+
+  constructor(data: OrganizationDomainJSON) {
+    super();
+    this.fromJSON(data);
+  }
+
+  static async create(organizationId: string, { name }: { name: string }): Promise<OrganizationDomainResource> {
+    const json = (
+      await BaseResource._fetch<OrganizationDomainJSON>({
+        path: `/organizations/${organizationId}/domains`,
+        method: 'POST',
+        body: { name } as any,
+      })
+    )?.response as unknown as OrganizationDomainJSON;
+    return new OrganizationDomain(json);
+  }
+
+  update = (params: UpdateOrganizationDomainParams): Promise<OrganizationDomainResource> => {
+    return this._basePatch({
+      method: 'PATCH',
+      path: `/organizations/${this.organizationId}/domains/${this.id}`,
+      body: params,
+    });
+  };
+
+  delete = (): Promise<void> => {
+    return this._baseDelete({
+      path: `/organizations/${this.organizationId}/domains/${this.id}`,
+    });
+  };
+
+  // static async get({
```

Comment:
❓ Do we need this or we can drop it?

Reply:
Already dropped, maybe it was visible in the per commit review